src/containers/search-bar.js

```diff
@@ -0,0 +1,131 @@
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchForecast, setDefaultCity, fetchWithCords } from "../actions/index";
+import axios from "axios";
+
+const SearchBar = () => {
+  const currCity = useSelector(state => {
+    return state.forecast.forecast_data.city});
+
+  const default_city = useSelector(state => {
+    return state.default_city});
+  
+
+  const errors = useSelector((state) => {
+    return state.forecast.error});
+  
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (default_city) {
+      dispatch(fetchForecast(default_city));
+    }
+  }, []);
+
+  const onSearchCity = (e) => {
+    e.preventDefault();
+    const city = e.target.elements.city.value
+    dispatch(fetchForecast(city));
+  }
+
+  const onSetDefaultCity = () => {
+    if(!errors){
+      dispatch(setDefaultCity(currCity));
+    }
+  }
+
+  const geoSuccess = function(position) {
+    var latitude  = position.coords.latitude;
+    var longitude = position.coords.longitude;
+
+    const url = `https://api.openweathermap.org/geo/1.0/reverse?lat=${latitude}&lon=${longitude}&appid=d8962aecd5213db8fadda59d04d64a81`;
+
+    axios.get(`${url}`)
+    .then(function (response) {
+      // handle success
+      dispatch(fetchForecast(response.data[0].name))
+      loadAndUnloadPage("loaded", false);
+    })
+  };
+  
+  const geoError = function(postition) {
+    console.log("geo error");
+  }
+
+  // add loading and disable buttons or opposite of that
+  const loadAndUnloadPage = function(action, state) {
+    if (action === "loading") {
+      document.getElementById('root').classList.add('loading');
+    } else {
+      document.getElementById('root').classList.remove('loading')
+    }
```

Comment:
There are simpler ways to accomplish this; would recommend handling this state at the highest level of components where it is relevant (you could potentially handle loading state here if all you want to do is show the loading cursor)